Package.swift

```diff
@@ -63,7 +63,7 @@ let package = Package(
         "OrderedCollections",
       ],
       path: "Sources/Collections",
-      exclude: ["CMakeLists.txt"],
+      exclude: ["CMakeLists.txt", "Collections.docc"],
```

Comment:
@finestructure Would adding DocC bundle to the exclude list (because they generate warnings in older versions of Swift and we don't want to add versioned `Package.swift` just yet) interfere with hosting on SwiftPackageIndex.com?

Reply:
I've checked the tip of this PR and docc still generates the docs regardless, so this won't be a problem:

```
========================================
Checkout
========================================
Clone URL: https://github.com/apple/swift-collections
Reference: b4c1070741a119654864b736268c3f03f6799a1b
<- snip ->
========================================
GenerateDocs
========================================
Generating docs at path:  /var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AD3353BD-9650-4E82-884E-8983E40CDE2E/checkout/.docs/apple/swift-collections/b4c1070741a119654864b736268c3f03f6799a1b
Repository:               apple/swift-collections
Swift version used:       5.7
Target:                   Collections
WARNING: environment variable API_BASEURL is not set
Fetching https://github.com/apple/swift-docc-plugin from cache
Fetched https://github.com/apple/swift-docc-plugin (0.74s)
Computing version for https://github.com/apple/swift-docc-plugin
Computed https://github.com/apple/swift-docc-plugin at 1.0.0 (0.48s)
Creating working copy for https://github.com/apple/swift-docc-plugin
Working copy of https://github.com/apple/swift-docc-plugin resolved at 1.0.0
Building for debugging...
<- snip ->
Build complete! (1.06s)
Generating documentation for 'Collections'...
Converting documentation...
Conversion complete! (0.03s)
Generated DocC archive at '/private/var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AD3353BD-9650-4E82-884E-8983E40CDE2E/checkout/.docs/apple/swift-collections/b4c1070741a119654864b736268c3f03f6799a1b'
WARNING: environment variable API_BASEURL is not set
Target:                   DequeModule
WARNING: environment variable API_BASEURL is not set
Building for debugging...
Build complete! (0.09s)
Generating documentation for 'DequeModule'...
Converting documentation...
Conversion complete! (0.15s)
Generated DocC archive at '/private/var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AD3353BD-9650-4E82-884E-8983E40CDE2E/checkout/.docs/apple/swift-collections/b4c1070741a119654864b736268c3f03f6799a1b'
Successfully merged into /var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AD3353BD-9650-4E82-884E-8983E40CDE2E/checkout/.docs-staging/apple/swift-collections/b4c1070741a119654864b736268c3f03f6799a1b/index/index.json
WARNING: environment variable API_BASEURL is not set
Target:                   OrderedCollections
WARNING: environment variable API_BASEURL is not set
Building for debugging...
Build complete! (0.09s)
Generating documentation for 'OrderedCollections'...
Converting documentation...
Conversion complete! (0.57s)
Generated DocC archive at '/private/var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AD3353BD-9650-4E82-884E-8983E40CDE2E/checkout/.docs/apple/swift-collections/b4c1070741a119654864b736268c3f03f6799a1b'
Successfully merged into /var/folders/nk/tlpts6bs799038htr6rx1xnc0000gn/T/AD3353BD-9650-4E82-884E-8983E40CDE2E/checkout/.docs-staging/apple/swift-collections/b4c1070741a119654864b736268c3f03f6799a1b/index/index.json
```

Reply:
Hold on, that doesn't necessarily mean the docc bundle and its articles would be included. I'll confirm.

Reply:
Yes, this does indeed strip out the landing page:

<img width="1148" alt="CleanShot 2022-12-07 at 09 20 44@2x" src="https://user-images.githubusercontent.com/65520/206126587-b06c0a61-c5d4-442a-92fe-9c9abcd8b6eb.png">

vs without the exclude:

<img width="1148" alt="CleanShot 2022-12-07 at 09 21 17@2x" src="https://user-images.githubusercontent.com/65520/206126625-8447ebb2-79ae-4978-b10b-590a5a978afe.png">

Generated via

```
swift package --disable-sandbox plugin preview-documentation --target Collections
```

Reply:
A hacky workaround over adding a versioned manifest could be adding this at the end of the manifest, per target:

```swift
#if swift(<5.7)
for target in package.targets.filter({ ["Collections" /* extend as needed */ ].contains($0.name) }) {
    package.targets[0].exclude = ["CMakeLists.txt", "\(target.name).docc"]
}
#endif
```

Reply:
Hm, that workaround looked promising at first, but it would still generate unhandled file warnings in 5.7+.

I think I'll just have to add a versioned manifest for SPMs ≧ 5.5 -- that will need a bit of more validation testing before I tag a new release, but it'll work!

Reply:
Thanks @finestructure @lorentey!